bin/create_vembrane_fields.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python
+
+import re
+import sys
+import argparse
+
+
+def format_string(input_string, column_name, fields_with_sampleindex=None, sampleindex=None, header_prefix=None):
+    if fields_with_sampleindex is None and sampleindex is not None:
+        fields_with_sampleindex = "all"
+
+    def field_format(match):
+        field_name = match.group(1)
+        if header_prefix is not None:
+            return _format_header(field_name, header_prefix)
+        elif fields_with_sampleindex == "all":
+            if sampleindex is not None:
+                return _format_field(field_name, sampleindex)
+            else:
+                raise ValueError("Sample index needs to be specified when using fields_with_sampleindex == all.")
+        elif fields_with_sampleindex and field_name in fields_with_sampleindex is not None:
+            if sampleindex is not None:
+                return _format_field(field_name, sampleindex)
+            else:
+                raise ValueError("Sample index needs to be specified when using fields_with_sampleindex.")
+        else:
+            return _format_field(field_name)
+
+    def _format_field(field, sampleindex=None):
+        result = f'{column_name}["{field}"]'
+        if sampleindex is not None:
+            result += f"[{sampleindex}]"
+        return result
+
+    def _format_header(field, header_prefix):
+        result = f"{header_prefix}_{field}"
+        return result
+
+    formatted_string = re.sub(r"([A-Za-z_]+)", field_format, input_string)
```

Comment:
this can return the input_string again if the pattern wasnt found, which is likely unintended
we can validate the input, through for example a regex that checks and rejects all unusual input_strings here

Reply:
I am not sure what you meant. The re.sub() functions are used to replace strings but keep e.g. mathematical operands, brackets, etc. since vembrane can parse those and allows for example easier allele fraction calculation.

Reply:
this is fixed now. Since the input is checked now, bad inputs wont pass through this anymore